### PR TITLE
Fix stuff

### PR DESCRIPTION
**Pull request by ShubhankarKG**
1. Scripts for starting server, client, and both concurrently added.
2. .env.sample for dev.
3. Refactor code in respective folders. (To be followed in a later commit as well).
4. introduce typescript support for both server and client.
5. Prepare folder structure for server and added few more scripts for build.
6. Sample test added.

### CODE REVIEW

**Comment by ShubhankarKG** [comment]
I feel this amount of setup is good to go, I would like to hear your thoughts on this @AliabbasMerchant .

File: client/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (AliabbasMerchant) [inline client/tsconfig.json:3]: Shouldn't we target es6?

File: server/src/config/db.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import dotenv from "dotenv";
+import mongoose from "mongoose";
+
+dotenv.config();
```
Comment (AliabbasMerchant) [inline server/src/config/db.ts:4]: I believe we dont need to do this again, if we have already configured it in app.ts

File: server/src/router.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const router = express.Router();
+
+router.get("/", (req, res) => {
+  res.status(200).json({ message: "YO" });
+});
+
+if (process.env.NODE_ENV === "production") {
+  router.get("*", (req, res) => {
+    res.sendFile(__dirname + "/client/build/index.html");
```
Comment (AliabbasMerchant) [inline server/src/router.ts:11]: The whole program will need to be started from the root of the repo folder, right, for this to work? (Doesn't matter as such, just confirming)

**Review by AliabbasMerchant — COMMENTED** [review]
Looks good to me! Thank you for your efforts!

File: server/src/router.ts (hunk 3 above)
Comment (ShubhankarKG) [reply server/src/router.ts:11]: Thanks, good catch. I used router in `/api`, so this would have been misleading.

**Comment by ShubhankarKG** [comment]
Merging this one then.

**Pull request merged**